### PR TITLE
dev/core#4429 Fix checking to see if a price option is able to be use…

### PR DESCRIPTION
**Pull request by seamuslee001**
…d for CiviMember or CiviEvent

Overview
----------------------------------------
This fixes an issue because extends is now a declared searlized field

Before
----------------------------------------
For select fields membership defaults may not show up or event defaults

After
----------------------------------------
Correct fields show up

ping @eileenmcnaughton @petednz

### CODE REVIEW

File: CRM/Price/Form/Option.php [hunk 1]
```diff
@@ -79,10 +79,9 @@ public function setDefaultValues() {
       }
     }
 
-    $memberComponentId = CRM_Core_Component::getComponentID('CiviMember');
     $extendComponentId = CRM_Core_DAO::getFieldValue('CRM_Price_DAO_PriceSet', $this->_sid, 'extends', 'id');
 
-    if (!isset($defaults['membership_num_terms']) && $memberComponentId == $extendComponentId) {
+    if (!isset($defaults['membership_num_terms']) && $this->isComponentPriceOption($memberComponentId, 'CiviMember')) {
```
Comment (demeritcowboy) [inline CRM/Price/Form/Option.php:84]: The `$memberComponentId` var is removed just above?

File: CRM/Price/Form/Option.php [hunk 2]
```diff
@@ -344,4 +341,19 @@ public function postProcess() {
...
   }
 
+  /**
+   * Check to see if this is within a price set that supports the specific component
+   * @var string $extends separator encoded string of Component ids
+   * @var string $component Component name
+   *
+   * @return bool
+   */
+  private function isComponentPriceOption(string $extends, string $comonent): bool {
```
Comment (demeritcowboy) [inline CRM/Price/Form/Option.php:351]: `$comonent`

File: CRM/Price/Form/Option.php (hunk 1 above)
Comment (eileenmcnaughton) [reply CRM/Price/Form/Option.php:84]: right it should be `$extendComponentId` - athought I'd probably move that look up into the function

File: CRM/Price/Form/Option.php (hunk 2 above)
Comment (seamuslee001) [reply CRM/Price/Form/Option.php]: Fixed thanks @demeritcowboy

File: CRM/Price/Form/Option.php (hunk 1 above)
Comment (seamuslee001) [reply CRM/Price/Form/Option.php]: Fixed thanks @demeritcowboy

**Pull request merged**